Addressing some documentation issues with annotations

mantognini opened the pull request:
Related to the discussion in #98.

I used `private[stainless]` to disallow `@ignore` in user code as it seems `@extern` should be used instead. The remaining questions are, should we get rid of `@traceInduct`, `@internal` and `@pure` since they don't seem well tested at best? Or how do you see things, @samarion ?

mantognini commented:
An additional question: synthesis being not yet implemented in stainless, should I remove this part of the documentation & library as well for now? (No tests rely on these features ATM.)

samarion commented:
Yes, I believe that removing the synthesis stuff makes sense. We can also remove (or comment?) the `@traceInduct`, `@internal` and `@pure` annotations though I think `@traceInduct` at least should make its way back into Stainless at some point.

mantognini commented:
Okay, so I've removed the synthesis part (`holes` trees and transformers still remains though) and I also removed the three annotations as they weren't handled anyway. Those separate commits can be reverted in the future if needed.
